src/physicsobjects.hxx

```diff
@@ -364,7 +365,53 @@ auto PtCorrection(auto &df, const std::string &corrected_pt,
         df.Define(corrected_pt, tau_pt_correction_lambda, {pt, decayMode});
     return df1;
 }
-
+/// Function to correct tau pt with correctionlib
+///
+/// \param[in] df the input dataframe
+/// \param[out] corrected_pt name of the corrected tau pt to be calculated
+/// \param[in] pt name of the raw tau pt
+/// \param[in] eta name of raw tau eta
+/// \param[in] decayMode decay mode of the tau
+/// \param[in] genMatch column with genmatch values (from prompt e, prompt mu,
+/// tau->e, tau->mu, had. tau) \param[in] sf_file path to the file with the tau
+/// energy correction \param[in] variation name of the tau energy correction
+/// variation ("nom","up","down")
+///
+/// \return a dataframe containing the new mask
+auto PtCorrection(auto &df, const std::string &corrected_pt,
+                  const std::string &pt, const std::string &eta,
+                  const std::string &decayMode, const std::string &genMatch,
+                  const std::string &sf_file, const std::string &variation,
+                  const std::vector<int> &SelectedDMs) {
+    auto evaluator =
+        correction::CorrectionSet::from_file(sf_file)->at("tau_energy_scale");
```

Comment:
should we turn `tau_energy_scale` into a config parameter, if the taupog descides to change thier file naming ?

Reply:
Done